File: ООП Lab1.3/Lab1.3/String.cs
using System;

namespace StringOperations
{
    public class CustomString
    {
        private string value;

        public CustomString()
        {
            value = "";
        }

        public CustomString(string val)
        {
            value = val;
        }

        public CustomString(CustomString other)
        {
            value = other.value;
        }

        public int Length()
        {
            return value.Length;
        }

        public string GetValue()
        {
            return value;
        }

        public static CustomString operator +(CustomString a, CustomString b)
        {
            return new CustomString(a.value + b.value);
        }

        public static CustomString operator *(CustomString a, int times)
        {
            string result = "";
            for (int i = 0; i < times; i++)
            {
                result += a.value;
            }
            return new CustomString(result);
        }
    }
}

File: Lab2.1/String.cpp
#include "String.h"

String::String() : value("") {}
String::String(const string& val) : value(val) {}
String::String(const String& other) : value(other.value) {}

int String::length() const {
	return value.length();
}

std::string String::getValue() const {
	return value;
}